src/reducers/vis-state-updaters.js

```diff
@@ -1124,6 +1125,32 @@ export const showDatasetTableUpdater = (state, action) => {
   };
 };
 
+/**
+ * When user clicks on colored square ..............
+ * @memberof visStateUpdaters
+ * @type {typeof import('./vis-state-updaters').updateDatasetColorUpdater}
+ * @public
+ */
+export const updateDatasetColorUpdater = (state, action) => {
+  const {dataId, newColor} = action;
+  const {datasets} = state;
+
+  if (newColor) {
+    const existing = datasets[dataId];
+    existing.updateDatasetColor(newColor);
+    const datasetNumber = state.datasetUpdateNumbers[dataId] || 0;
+    // @ts-ignore
+    return {
```

Comment:
use `copyTableAndUpdate` to update dataset, you won't need `datasetUpdateNumbers`